validation.go

```diff
@@ -193,27 +191,33 @@ func convertValidations(validations []Validation) []cpluginv1.ParameterValidatio
 	return out
 }
 
-// ApplyConfigValidations a utility function that applies all the validations for parameters, return an error that
+// private struct to group all validation functions
+type validator map[string]Parameter
+
+// applyConfigValidations is a utility function that applies all the validations for parameters, returns an error that
 // consists of a combination of errors from the configurations.
-func applyConfigValidations(params map[string]Parameter, config map[string]string) error {
+func (v validator) applyConfigValidations(config map[string]string) error {
```

Comment:
Now that this is a method on a private struct we can make it public - this makes it clear it is supposed to be called by some other code and not the validator, while other private methods are only internal helpers. And I think we can just call it `Validate` if you ask me, keep it simple 🙂 
```suggestion
func (v validator) Validate(config map[string]string) error {
```